tracekit.js

```diff
@@ -118,27 +118,38 @@ TraceKit.report = (function reportModuleWrapper() {
         lastException = null,
         lastExceptionStack = null;
 
+    function isWindowAccessible(win) {
+      try {
+        return (win.location.host);
+      } catch (e) {}
+    }
     /**
      * Add a crash handler.
      * @param {Function} handler
      * @memberof TraceKit.report
      */
-    function subscribe(handler) {
-        installGlobalHandler();
-        handlers.push(handler);
+    function subscribe(handler, win) {
+      win = (win || window);
+      if (isWindowAccessible(win)) {
+        TraceKit.windowPointer = win;
+        installGlobalHandler(handler, win);
+      }
     }
 
     /**
      * Remove a crash handler.
      * @param {Function} handler
      * @memberof TraceKit.report
      */
-    function unsubscribe(handler) {
+    function unsubscribe(handler, win) {
+      win = (win || window);
+      if (isWindowAccessible(win)) {
         for (var i = handlers.length - 1; i >= 0; --i) {
-            if (handlers[i] === handler) {
+          if (handlers[i][0] === handler && win === handlers[i][1]) {
```

Comment:
We weren't doing this before, but since you are calling unsubscribe and you have an array with the previous error handler. Do you think we should be setting it back?